src/util/http/HttpClient.cpp

```diff
@@ -90,10 +91,8 @@ std::istringstream HttpClientImpl<StreamType>::sendRequest(
     const boost::beast::http::verb& method, std::string_view host,
     std::string_view target, std::string_view requestBody,
     std::string_view contentTypeHeader, std::string_view acceptHeader) {
-  // Check that we have a stream (obtained via a call to `openStream` above).
-  if (!stream_) {
-    throw std::runtime_error("Trying to send request without connection");
-  }
+  // Check that we have a stream (created in the constructor).
+  AD_CONTRACT_CHECK(stream_);
```

Comment:
Is there a way to make this fail on purpose?
I think otherwise it is an `AD_CORRECTNESS_CHECK`.